src/symbolic_simulate.py

```diff
@@ -288,11 +288,13 @@ def calculate_edp(self, hw):
             hw, self.execution_time
         )
 
+        print(f"total passive energy: {self.total_passive_energy}", flush=True)
         self.edp = self.execution_time * (self.total_active_energy + self.total_passive_energy)
 
         self.execution_time = self.execution_time.xreplace(cacti_subs)
         self.total_active_energy = self.total_active_energy.xreplace(cacti_subs)
         self.total_passive_energy = self.total_passive_energy.xreplace(cacti_subs)
+        self.edp = self.execution_time * (self.total_active_energy + self.total_passive_energy)
```

Comment:
Actually if you log things a bit closely. this value (after evaluating E_p, E_a and D and then multiplying) doesn't agree with the edp you get if you combine the expressions, and then evaluate. 
This is also something important to look at